### PR TITLE
Add PrintHelp method to API feature

### PR DESCRIPTION
**Pull request by quantumagi**
This PR adds a PrintHelp method to the API feature to display command-line help related to the API feature when required.

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Api/ApiSettings.cs [hunk 1]
```diff
@@ -97,5 +90,39 @@ public void Load(NodeSettings nodeSettings)
...
         }
+
+        /// <summary>
+        /// Determines the default API port.
+        /// </summary>
+        /// <param name="network">The network to use.</param>
+        /// <returns>The default API port.</returns>
+        private static int DefaultPort(Network network)
+        {
+            int port;
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Features.Api/ApiSettings.cs]: you can do without the variable and just `return network.IsTest() ? TestBitcoinApiPort : DefaultBitcoinApiPort;` etc

File: src/Stratis.Bitcoin.Features.Api/ApiSettings.cs [hunk 2]
```diff
@@ -97,5 +90,39 @@ public void Load(NodeSettings nodeSettings)
 
             this.callback?.Invoke(this);
         }
+
+        /// <summary>
+        /// Determines the default API port.
+        /// </summary>
+        /// <param name="network">The network to use.</param>
+        /// <returns>The default API port.</returns>
+        private static int DefaultPort(Network network)
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Features.Api/ApiSettings.cs]: Change the name to GetDefaultPort(...)

File: src/Stratis.Bitcoin.Features.Api/ApiSettings.cs [hunk 3]
```diff
@@ -97,5 +90,39 @@ public void Load(NodeSettings nodeSettings)
...
+            }
+            else
+            {
+                port = network.IsTest() ? TestStratisApiPort : DefaultStratisApiPort;
+            }
+
+            return port;
+        }
+
+        /// <summary>Prints the help information on how to configure the rpc settings to the logger.</summary>
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Features.Api/ApiSettings.cs]: API settings

File: src/Stratis.Bitcoin.Features.Api/ApiSettings.cs [hunk 4]
```diff
@@ -97,5 +90,39 @@ public void Load(NodeSettings nodeSettings)
...
+        /// <param name="network">The network to use.</param>
+        public static void PrintHelp(Network network)
+        {
+            var defaults = NodeSettings.Default();
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Features.Api/ApiSettings.cs]: you can move the `var defaults` to the bottom, closer to where it's actually being used.

File: src/Stratis.Bitcoin.Features.Api/ApiSettings.cs (hunk 1 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.Api/ApiSettings.cs]: Done.

File: src/Stratis.Bitcoin.Features.Api/ApiSettings.cs (hunk 2 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.Api/ApiSettings.cs]: Done.

File: src/Stratis.Bitcoin.Features.Api/ApiSettings.cs (hunk 3 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.Api/ApiSettings.cs]: Done.

File: src/Stratis.Bitcoin.Features.Api/ApiSettings.cs (hunk 4 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.Api/ApiSettings.cs]: Done.

**Comment by bokobza** [comment]
Lovely, thank you.

**Pull request merged**